### PR TITLE
JSONB, is not being set to its default value

### PR DESCRIPTION
**Pull request by msvinaykumar**
The potential issue is that after an increase in database size, the DataSource column, which is of type JSONB, is not being set to its default value. However, it allows for this if we perform a database cleanup. Therefore, the solution is to set the default value of the JSONB-type column to null.   

created issue for same

https://github.com/kruize/autotune/issues/1065

### CODE REVIEW

**Comment by msvinaykumar** [comment]
@chandrams this issue not able to reproduce locally  as it requires DB size increase to existing DB without clean up command.

File: migrations/kruize_experiments_ddl.sql [hunk 1]
```diff
@@ -1,7 +1,7 @@
...
-create table IF NOT EXISTS kruize_results (interval_start_time timestamp(6) not null, interval_end_time timestamp(6) not null, experiment_name varchar(255) not null, cluster_name varchar(255) , duration_minutes float(53) not null, extended_data jsonb, meta_data jsonb, version varchar(255), primary key (experiment_name, interval_end_time, interval_start_time)) PARTITION BY RANGE (interval_end_time);
+create table IF NOT EXISTS kruize_recommendations (interval_end_time timestamp(6) not null, experiment_name varchar(255) not null, cluster_name varchar(255), extended_data jsonb , version varchar(255), primary key (experiment_name, interval_end_time)) PARTITION BY RANGE (interval_end_time);
+create table IF NOT EXISTS kruize_results (interval_start_time timestamp(6) not null, interval_end_time timestamp(6) not null, experiment_name varchar(255) not null, cluster_name varchar(255) , duration_minutes float(53) not null, extended_data jsonb, meta_data jsonb DEFAULT NULL, version varchar(255), primary key (experiment_name, interval_end_time, interval_start_time)) PARTITION BY RANGE (interval_end_time);
```
Comment (chandrams) [inline migrations/kruize_experiments_ddl.sql:4]: Do we need to set DEFAULT NULL or extended_data jsonb as well?

Comment (msvinaykumar) [reply migrations/kruize_experiments_ddl.sql:4]: no extended_data cannot be set  DEFAULT NULL, It should have values always.

**Review by chandrams — APPROVED** [review]
LGTM

File: src/main/java/com/autotune/database/helper/DBHelpers.java [hunk 2]
```diff
@@ -312,6 +314,7 @@ public static KruizeResultsEntry convertExperimentResultToExperimentResultsTable
                             Double.valueOf((experimentResultData.getIntervalEndTime().getTime() -
                                     experimentResultData.getIntervalStartTime().getTime()) / (60 * 1000))
                     );
+                    kruizeResultsEntry.setMeta_data(null);
```
[the remark this answers is not in the document]
Comment (msvinaykumar) [reply src/main/java/com/autotune/database/helper/DBHelpers.java:317]: Yes, however, the 'datasource' element has not been implemented at the code level; it was only conceptualized during the design phase. To verify the existence of the datasource, we would need to declare that variable in the API object and handle JSON parsing accordingly. All these modifications can be implemented once the datasource is active. Currently, we have chosen to minimize changes to focus on bug fixes.

**Comment by msvinaykumar** [comment]
merged into MVP 
https://github.com/kruize/autotune/pull/1085

**Pull request closed without merging**